File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        result = defaultdict(list)
        
        for str in strs:
            sorted_key = ''.join(sorted(str))
            # if sorted_key not in result:
            #     result[sorted_key] = []
            result[sorted_key].append(str)
        
        return list(result.values())
            

File: 0049-group-anagrams/0049-group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    const result = {};

    for (const str of strs) {
        const sortedKey = str.split('').sort().join('');
        if (!result[sortedKey]) {
            result[sortedKey] = [];
        }
        result[sortedKey].push(str);
    }

    return Object.values(result);
};